runner/entrypoint.sh

```diff
@@ -66,5 +66,10 @@ for f in runsvc.sh RunnerService.js; do
   sudo mv {patched,bin}/${f}
 done
 
+args=()
+if [ "${RUNNER_EPHEMERAL}" == "true" ]; then
```

Comment:
I think this breaks newly released runner images to not work with previous versions of actions-runner-controller.

I'm not saying we must support all the previous actions-runner-controller releases, but I'd love it if it could support one or two previous versions.

That being said, can we probably make this backward-compatible by changing the conditional to:

```suggestion
if [ "${RUNNER_EPHEMERAL}" != "true" ]; then
```

WDYT?

Reply:
It is backward-compatible as per what I tested, [this](https://github.com/actions-runner-controller/actions-runner-controller/pull/498/files#diff-f4c239205beda2ef5d9d041809cd93cb32958b6b658fb0b2331c85b853777465R509) condition evaluates the variable to `true` when the field isn't mentioned.

Reply:
@thejasn Yes, it does seem to be backward-compatible in terms of Runner CRD changes.

What I was saying was though newer runner images being deployed by older actions-runner-controller and older CRDs.

Unlike the controller, [runner images are continuously built and released from our default branch](https://github.com/actions-runner-controller/actions-runner-controller/blob/master/.github/workflows/build-and-release-runners.yml), without waiting for a controller release. That implies an older version of actions-runner-controller may use newer runner images depending on the configuration on the user's side.

If the older version of actions-runner-controller instructed to use the latest runner image build after this change,  wouldn't the runner pod gets an empty `RUNNER_EPHEMERAL` which results in missing `--once`?

Reply:
Understood, didn't know runner would be released independently of the controller.